src/Repeater/Repeater.ts

```diff
@@ -0,0 +1,61 @@
+import { Protocol } from '../RequestExecutor';
+
+export interface RepeaterEvent<I, O = void> {
+  request: I;
+  response: O;
+}
+
+export type RepeaterRequestEvent = RepeaterEvent<
+  {
+    protocol: Protocol;
+    url: string;
+    method?: string;
+    headers?: Record<string, string | string[]>;
+    correlationIdRegex?: string;
+    body?: string;
+  },
+  {
+    protocol: Protocol;
+    statusCode?: number;
+    message?: string;
+    errorCode?: string;
+    headers?: Record<string, string | string[] | undefined>;
+    body?: string;
+  }
+>;
+
+export type RepeaterDeployedEvent = RepeaterEvent<
+  undefined,
+  {
+    repeaterId: string;
+  }
+>;
+
+export type RepeaterReconnectionFailedEvent = RepeaterEvent<{
+  error: Error;
+}>;
+
+export interface RepeaterEvents {
+  deployed: RepeaterDeployedEvent;
+  request: RepeaterRequestEvent;
+  reconnection_failed: RepeaterReconnectionFailedEvent;
+}
+
+export type RepeaterEventHandler<E extends keyof RepeaterEvents> = (
+  payload: RepeaterEvents[E]['request']
+) => RepeaterEvents[E]['response'] | Promise<RepeaterEvents[E]['response']>;
+
+export interface Repeater {
```

Comment:
Instead of introducing the new abstraction with a superficial contract (i.e., `on`), the recommended approach is to introduce a new implementation of `RepeaterLauncher` instead. This way, both implementations can be retained until we have finalized the requirements.

Reply:
I added new `ServerRepeaterLauncher`. I also kept `Repeater` class (but renamed it to `RepeaterServer`)